src/cache/key-manifest.ts

```diff
@@ -0,0 +1,60 @@
+import { CACHE_KEY_PREFIX, ICache, KeyManifestEntry } from './shared';
+
+export class CacheKeyManifest {
+  private readonly manifestKey: string;
+
+  constructor(private cache: ICache, private clientId: string) {
+    this.manifestKey = this.createManifestKeyFrom(clientId);
+  }
+
+  async add(key: string): Promise<void> {
+    const existingEntry = await this.cache.get<KeyManifestEntry>(
```

Comment:
Since this is a unique list, a `Set` would be good here too:

```js
const keys = new Set(await this.cache.get(key) || []);
keys.add(key);
await this.cache.set({ keys: [...keys] })
```